packages/notifier/src/publish-kit.js

```diff
@@ -12,117 +11,64 @@ const { Fail, quote: q } = assert;
-    },
-  });
-};
-
-/**
- * Given a local or remote subscriber, returns a local AsyncIterable which
- * provides "prefix lossy" iterations of the underlying PublicationList.
- * By "prefix lossy", we mean that you may miss everything published before
- * you ask the returned iterable for an iterator. But the returned iterator
- * will enumerate each thing published from that iterator's starting point.
- *
- * If the underlying PublicationList is terminated, that terminal value will be
- * reported losslessly.
- *
- * @template T
- * @param {ERef<Subscriber<T>>} subscriber
- * @returns {AsyncIterable<T>}
- */
-export const subscribeEach = subscriber => {
-  const iterable = Far('EachIterable', {
-    [Symbol.asyncIterator]: () => {
-      const pubList = E(subscriber).subscribeAfter();
-      return makeEachIterator(pubList);
-    },
-  });
-  return iterable;
-};
-harden(subscribeEach);
-
-/**
- * @template T
- * @param {ERef<Subscriber<T>>} subscriber
- * @returns {AsyncIterator<T>}
- */
-const makeLatestIterator = subscriber => {
-  let latestPublishCount;
-  return Far('LatestIterator', {
-    next: () => {
-      const pubList = E(subscriber).subscribeAfter(latestPublishCount);
-      // Without an onRejection, if pubList is a rejection, this will
-      // propagate the rejection as it should.
-      return E.when(pubList, ({ head, publishCount }) => {
-        latestPublishCount = publishCount;
-        return head;
-      });
-    },
-  });
-};
-
-/**
- * Given a local or remote subscriber, returns a local AsyncIterable which
- * provides "lossy" iterations of the underlying PublicationList.
- * By "lossy", we mean that you may miss any published state if a more
- * recent published state can be reported instead.
- *
- * If the underlying PublicationList is terminated, that terminal value will be
- * reported losslessly.
- *
- * @template T
- * @param {ERef<Subscriber<T>>} subscriber
- * @returns {AsyncIterable<T>}
- */
-export const subscribeLatest = subscriber => {
-  const iterable = Far('LatestIterable', {
-    [Symbol.asyncIterator]: () => makeLatestIterator(subscriber),
-  });
-  return iterable;
-};
-harden(subscribeLatest);
-
 export const PublisherI = M.interface('Publisher', {
   publish: M.call(M.any()).returns(),
   finish: M.call(M.any()).returns(),
   fail: M.call(M.any()).returns(),
 });
+export const UpdateCountI = M.or(M.bigint(), M.number(), M.undefined());
 export const SubscriberI = M.interface('Subscriber', {
   subscribeAfter: M.call().optional(M.bigint()).returns(M.promise()),
+  getUpdateSince: M.call().optional(UpdateCountI).returns(M.promise()),
 });
 export const publishKitIKit = harden({
   publisher: PublisherI,
   subscriber: SubscriberI,
 });
 
+/**
+ * @template {object} Arg
+ * @template Ret
+ * @param {(arg: Arg) => Ret} fn
+ * @returns {(arg: Arg) => Ret}
+ */
+const weakMemoizeUnary = fn => {
+  const cache = new WeakMap();
+  return arg => {
+    /** @type {object} */
+    const oarg = arg;
```

Comment:
Can `oarg` be factored out?
```suggestion
  /** @type {(arg: Arg) => Ret} */
  return arg => {
```

Reply:
It exists only to appease the type constraints imposed by TS's typing of WeakMap.  Other solutions I played with involved significantly more noise.